components/autocomplete/index.js

```diff
@@ -330,11 +338,25 @@ export class Autocomplete extends Component {
 		// create a regular expression to filter the options
 		const search = open ? new RegExp( '(?:\\b|\\s|^)' + escapeRegExp( query ), 'i' ) : /./;
 		// filter the options we already have
-		const filteredOptions = open ? filterOptions( search, this.state[ 'options_' + open.idx ] ) : [];
+		let filteredOptions = open ? filterOptions( search, this.state[ 'options_' + open.idx ] ) : [];
 		// check if we should still suppress the popover
 		const suppress = ( open && wasSuppress === open.idx ) ? wasSuppress : undefined;
 		// update the state
 		if ( wasOpen || open ) {
+			if ( open ) {
+				if ( this.props.completers[ open.idx ].setSearch ) {
+					this.props.completers[ open.idx ].setSearch( query, this.props.completers[ open.idx ].getOptions ).then( ( options ) => {
+						const keyedOptions = map( options, ( option, i ) => ( { ...option, key: open.idx + '-' + i } ) );
+						filteredOptions = filterOptions( this.state.search, keyedOptions );
+						const selectedIndex = filteredOptions.length === this.state.filteredOptions.length ? this.state.selectedIndex : 0;
```

Comment:
I know my code probably made a similar bad assumption but just because the lengths are the same the item at that position may not be the same - something that will probably be more noticeable with the new search code. It might be desirable to search the filtered options?

Reply:
yea, good point, the slug should be unique.  i'll work on this some more.

Reply:
I removed my copy of this code block, the length check seems to work, I guess in some cases it might show the incorrect selection?